agent/xds/listeners_ingress.go

```diff
@@ -284,33 +360,68 @@ func makeSDSOverrideFilterChains(cfgSnap *proxycfg.ConfigSnapshot,
 	return chains, nil
 }
 
-func makeCommonTLSContextFromSDS(sdsCfg structs.GatewayTLSSDSConfig) *envoy_tls_v3.CommonTlsContext {
+var envoyTLSVersions = map[types.TLSVersion]envoy_tls_v3.TlsParameters_TlsProtocol{
+	types.TLSVersionAuto: envoy_tls_v3.TlsParameters_TLS_AUTO,
+	types.TLSv1_0:        envoy_tls_v3.TlsParameters_TLSv1_0,
+	types.TLSv1_1:        envoy_tls_v3.TlsParameters_TLSv1_1,
+	types.TLSv1_2:        envoy_tls_v3.TlsParameters_TLSv1_2,
+	types.TLSv1_3:        envoy_tls_v3.TlsParameters_TLSv1_3,
+}
+
+func makeTLSParametersFromGatewayTLSConfig(tlsCfg structs.GatewayTLSConfig) *envoy_tls_v3.TlsParameters {
+	tlsParams := envoy_tls_v3.TlsParameters{}
+
+	if tlsCfg.TLSMinVersion != types.TLSVersionUnspecified {
+		tlsParams.TlsMinimumProtocolVersion = envoyTLSVersions[tlsCfg.TLSMinVersion]
+	}
+	if tlsCfg.TLSMaxVersion != types.TLSVersionUnspecified {
+		tlsParams.TlsMaximumProtocolVersion = envoyTLSVersions[tlsCfg.TLSMaxVersion]
```

Comment:
Should probably explicitly handle not finding the string in the map.

Reply:
I _think_ that shouldn't be necessary at this point since the inputs have already been validated?